service/controller/v16/key/key.go

```diff
@@ -215,6 +215,23 @@ func IsPodDrained(pod *corev1.Pod) (bool, error) {
 	return b, nil
 }
 
+// ArePodContainersTerminated checks ContainerState for all containers present
+// in given pod. When all containers are in Terminated state, true is returned.
+func ArePodContainersTerminated(pod *corev1.Pod) bool {
+	var b bool
+	for _, cs := range pod.Status.ContainerStatuses {
+		if cs.State.Terminated != nil {
+			b = true
+		} else {
+			// If even single container within POD is not Terminated, return false.
+			b = false
+			break
+		}
+	}
+
+	return b
+}
```

Comment:
Isn't below simpler?

```
	for _, cs := range pod.Status.ContainerStatuses {
		if cs.State.Terminated == nil {
			return false
		}
	}

	return true
```

Reply:
This feels simpler.
But also this function returns true when all containers are **not** in Terminated state. I think we want to have the opposite here.

Reply:
`ArePodContainersTerminated` would indicate `true` when all containers are in Terminated state.

Reply:
Yes, Pawel's version is simpler and works correctly. Only return `true` when **all** containers are `Terminated`.